### PR TITLE
DOC-7456: Add N1QL transactions support

### PR DESCRIPTION
**Pull request by simon-dew**
The following draft documentation is ready for review:

* [BEGIN TRANSACTION](https://docs-staging.couchbase.com/server/7.0/n1ql/n1ql-language-reference/begin-transaction.html)
* [SET TRANSACTION](https://docs-staging.couchbase.com/server/7.0/n1ql/n1ql-language-reference/set-transaction.html)
* [SAVEPOINT](https://docs-staging.couchbase.com/server/7.0/n1ql/n1ql-language-reference/savepoint.html)
* [ROLLBACK TRANSACTION](https://docs-staging.couchbase.com/server/7.0/n1ql/n1ql-language-reference/rollback-transaction.html)
* [COMMIT TRANSACTION](https://docs-staging.couchbase.com/server/7.0/n1ql/n1ql-language-reference/commit-transaction.html)

Docs issue: [DOC-7456](https://issues.couchbase.com/browse/DOC-7456)

### CODE REVIEW

**Comment by simon-dew** [comment]
Please note that `txid`, `tximplicit`, etc., will be documented as part of [DOC-7485](https://issues.couchbase.com/browse/DOC-7458)

**Comment by simon-dew** [comment]
- [x] BEGIN TRANSACTION example 1: invalid character in string literal

File: modules/n1ql/pages/n1ql-language-reference/rollback-transaction.adoc [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+:savepoint: xref:n1ql-language-reference/savepoint.adoc
+:commit-transaction: xref:n1ql-language-reference/commit-transaction.adoc
+:rollback-transaction: xref:n1ql-language-reference/rollback-transaction.adoc
+
+[abstract]
+The `ROLLBACK TRANSACTION` statement enables you to rollback a transaction.
+
+== Purpose
+
+The `SAVEPOINT` statement enables you to rollback an ACID transaction.
```
Comment (programmatix) [inline modules/n1ql/pages/n1ql-language-reference/rollback-transaction.adoc]: s/SAVEPOINT/ROLLBACK I think

Comment (simon-dew) [reply modules/n1ql/pages/n1ql-language-reference/rollback-transaction.adoc]: Done

**Comment by simon-dew** [comment]
Update links to staging docs

**Pull request merged**